tests/test_milvus_vector_db_service.py

```diff
@@ -14,224 +14,351 @@
-    with open(filepath, "r") as file:
-        collection_config = json.load(file)
-
-    mock_milvus_client_fixture.has_collection.return_value = has_collection
-    name = collection_config.pop("name")
-    milvus_service_fixture.create(name=name, overwrite=overwrite, **collection_config)
-
-    if overwrite:
-        if has_collection:
-            mock_milvus_client_fixture.drop_collection.assert_called_once()
-        else:
-            mock_milvus_client_fixture.drop_collection.assert_not_called()
-    else:
-        mock_milvus_client_fixture.drop_collection.assert_not_called()
-
-    mock_milvus_client_fixture.create_collection_with_schema.assert_called_once()
-
-
-def test_insert(milvus_service_fixture: MilvusVectorDBService, mock_milvus_client_fixture: MilvusClient):
-    data = [
-        {
-            "id": 1, "embedding": [0.1, 0.2, 0.3], "age": 30
-        },
-        {
-            "id": 2, "embedding": [0.4, 0.5, 0.6], "age": 25
-        },
-    ]
-    milvus_service_fixture.insert(name="test_collection", data=data)
-    mock_milvus_client_fixture.get_collection.assert_called_once()
-
-
-@pytest.mark.parametrize(
-    "insert_data, expected_exception",
-    [
-        ([], RuntimeError),  # Collection does not exist
-        ([], RuntimeError),  # Other error scenario
-    ])
-def test_insert_error(milvus_service_fixture: MilvusVectorDBService,
-                      mock_milvus_client_fixture: MilvusClient,
-                      insert_data: list,
-                      expected_exception: Exception):
-    mock_milvus_client_fixture.has_collection.return_value = False
-    with pytest.raises(expected_exception):
-        milvus_service_fixture.insert("non_existent_collection", data=insert_data)
-
-
-def test_insert_dataframe(milvus_service_fixture: MilvusVectorDBService, mock_milvus_client_fixture: MilvusClient):
-    mock_insert = Mock()
-    mock_milvus_client_fixture.get_collection.return_value.insert = mock_insert
-
-    data = cudf.DataFrame({
-        "id": [1, 2],
-        "embedding": [[0.1, 0.2, 0.3], [0.4, 0.5, 0.6]],
-        "age": [30, 25],
-    })
-
-    milvus_service_fixture.insert_dataframe(name="test_collection", df=data)
-    mock_insert.assert_called_once()
-
-
-search_test_cases = [
-    ("query", {
-        "bool": {
-            "must": [{
-                "vector": {
-                    "embedding": [0.1, 0.2, 0.3]
-                }
-            }]
-        }
-    }, "query_result"),
-    ("data", [{
-        "id": 1, "embedding": [0.1, 0.2, 0.3]
-    }, {
-        "id": 2, "embedding": [0.4, 0.5, 0.6]
-    }], "data_result"),
-    ("error_value", None, None),
-    ("error_value", None, []),
-]
-
-
-@pytest.mark.parametrize("test_type, query_or_data, expected_result", search_test_cases)
+@pytest.fixture(scope="module", name="milvus_service_fixture")
+def milvus_service(milvus_server_uri: str):
+    service = MilvusVectorDBService(uri=milvus_server_uri)
+    yield service
+
+
+def load_yaml(filename):
```

Comment:
Is it loading a JSON or YAML?

Reply:
It's a JSON file. Renamed the function to `load_json`